src/app/dashboard/Missions/CreateMission.js

```diff
@@ -0,0 +1,489 @@
+  },
+  titleHeader: {
+    color: "#3739B5",
+    fontSize: "42px",
+    margin: "18px 9px",
+    fontWeight: "bold",
+  },
+  inputHeader: {
+    color: "#3739B5",
+    fontWeight: "600",
+    fontSize: "26px",
+    margin: "15px 0",
+    textAlign: "left",
+  },
+  select: {
+    width: "50%",
+  },
+  form: {
+    padding: "10px",
+  },
+  optionalTagBox: {
+    display: "flex",
+    flexDirection: "row",
+  },
+  optionalTag: {
+    color: "grey",
+    occupancy: "0.5",
+    fontSize: "10px",
+    padding: "0 5px",
+    margin: "15px 0",
+  },
+  dualForm: {
+    display: "flex",
+    flexDirection: "row",
+    justifyContent: "space-between",
+  },
+  form2: {
+    margin: "10px",
+    width: "100%",
+  },
+  inputHeader2: {
+    textAlign: "left",
+    fontWeight: "600",
+    fontSize: "16px",
+    lineHeight: "24px",
+    padding: "5px",
+    marginBottom: "5px",
+  },
+  optionalTag2: {
+    color: "grey",
+    occupancy: "0.5",
+    lineHeight: "24px",
+    fontSize: "10px",
+    padding: "0 5px",
+    margin: "4px 0",
+  },
+  header2Icons: {
+    color: "#3739B5",
+    marginTop: "3px",
+  },
+  dualSelect: {
+    maxWidth: "100%",
+  },
+  dualText: {
+    margin: "10px 0",
+  },
+  bigTextField: {
+    heigh: "150px",
+  },
+  button: {
+    background: "#3739B5",
+    margin: "15px 0",
+    color: "white",
+    padding: "10px",
+    width: "100%",
+  },
+  sideBySideSelect: {
+    display: "flex",
+    margin: "10px",
+  },
+  numSelector: {
+    width: "100px",
+  },
+  selectCreateButton: {
+    textAlign: "left",
+    padding: "10px",
+    color: "#3739B5",
+    maxWidth: "50%",
+  },
+  inputLabel: {
+    fontSize: "20px",
+  },
+  menuItem: {
+    display: "flex",
+  },
+}));
+
+const CreateMission = () => {
+  const classes = useStyles();
+  const [missionType, setMissionType] = useState();
+  const [items, setItems] = useState();
+  const [missionFund, setMissionFund] = useState(1);
+  const [recipient, setRecipient] = useState({ name: "default", phoneNumber: "" });
+  const [volunteer, setVolunteer] = useState({ name: "default", phoneNumber: "" });
+  const [dropOff, setDropOff] = useState({ address: "", date: "" });
+  const [pickUp, setPickUp] = useState({ address: "", date: "" });
+  const [comment, setComment] = useState("");
+  // master mission object
+  const [newMissionObject, setNewMissionObject] = useState({});
+
+  // master submit
+  async function submitMission(e) {
+    e.preventDefault();
+    setNewMissionObject({
+      missionId: uuidv4(),
+      createdAt: new Date(),
+      createdBy: "organizer", // make dynamic
+      missionType: missionType,
+      associatedItems: items,
+      fund: missionFund,
+      recipient: recipient,
+      volunteer: volunteer,
+      dropOff: dropOff,
+      pickUp: pickUp,
+      comment: comment,
+    });
+    try {
+      const createdMission = await Mission.create(newMissionObject);
+      console.log(createdMission);
+    } catch (error) {
+      console.log(error);
```

Comment:
that is why I wanted providerContext with snackbar -> I can use that on the actually Mission.create , instead of having to add that component everywhere, some other days though, lets leave it for now

Reply:
Okay noted Ill leave it alone at the moment